adafruit_requests.py

```diff
@@ -54,22 +54,40 @@
 __version__ = "0.0.0-auto.0"
 __repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_Requests.git"
 
-_the_interface = None  # pylint: disable=invalid-name
-_the_sock = None  # pylint: disable=invalid-name
+# This module mirror CPython's socket class. It is settable because the network devices are external
+# to CircuitPython.
+_socket = None  # pylint: disable=invalid-name
 
 
-def set_socket(sock, iface=None):
-    """Helper to set the global socket and optionally set the global network interface.
-    :param sock: socket object.
-    :param iface: internet interface object
+def set_socket(socket):
```

Comment:
Could you edit the `simpletest` examples to reflect the removal of the `iface` kwarg? 

https://github.com/adafruit/Adafruit_CircuitPython_Requests/blob/master/examples/requests_simpletest.py#L32

Reply:
Also - why was the call to `set_interface` (https://github.com/adafruit/Adafruit_CircuitPython_ESP32SPI/blob/master/adafruit_esp32spi/adafruit_esp32spi_socket.py#L40) removed? Would setting the interface be handled in user-code, instead?

Reply:
Ok, examples updated.

Setting the interface is usually done by WifiManager now but is manual in these examples. Further into the future I think the socket should assume the interface instead. We should be able to isolate it to there in the future.

Ideally this library and others like MQTT should work with CPython's socket alone and then the BLE, ESP32SPI and ethernet libraries provide their own socket implementations.